components/button/test/button-icon.test.js

```diff
@@ -0,0 +1,31 @@
+import '../button-icon.js';
+import { expect, fixture, oneEvent } from '@open-wc/testing';
+
+const basicFixture = '<d2l-button-icon icon="d2l-tier1:gear" text="Icon Button"></d2l-button-icon>';
+const disabledFixture = '<d2l-button-icon disabled icon="d2l-tier1:gear" text="Icon Button"></d2l-button-icon>';
+
+describe('d2l-button-icon', () => {
+
+	it('dispatches click event when clicked', async() => {
+		const el = await fixture(basicFixture);
```

Comment:
Does this handle the waiting for `updateComplete` for us?  That's super nice

Reply:
It does indeed! Although I need to find a way to handle components using the localize mixin, as they need to wait a bit longer.

Reply:
Could wait for a localize event.  For composite/app components, they would not know how many events to wait for though.  The async container mixin pattern comes to mind, though it was not written with localization specifically in mind, though it is applicable.